src/views/PageSpotRequest.vue

```diff
@@ -0,0 +1,198 @@
+                <template v-slot="props">
+                    <div>
+                        {{ props.row.Address }}
+                    </div>
+                </template>
+            </b-table-column>
+
+            <!-- Status Column -->
+            <b-table-column field="Status" label="Status" width="150px">
+                <template v-slot="props">
+                    <div class="status-column">
+                        <div class="status-part">
+                            <span class="tag my-status">
+                                {{ getSpotRequestStatusLabel(props.row.Status) }}
+                            </span>
+                        </div>
+                    </div>
+                </template>
+            </b-table-column>
+
+            <b-table-column field="Remark" label="Remark" sortable cell-class="has-text-left">
+                <template v-slot="props">
+                    <div>
+                        {{ props.row.Remark }}
+                    </div>
+                </template>
+            </b-table-column>
+
+            <b-table-column field="LastCallDate" label="Last Call Date" sortable cell-class="has-text-left">
+                <template v-slot="props">
+                    <div>
+                        {{ props.row.LastCallDate}}
+                    </div>
+                </template>
+            </b-table-column>
+        </b-table>
+    </div>
+</template>
+
+<script>
+import { mapState, mapActions } from 'vuex';
+import LoaderModal from '../components/extras/LoaderModal.vue';
+import MoleculeSearchBox from '../components/molecules/MoleculeSearchBox.vue';
+import { getSpotRequestStatusLabel } from '../constant/enums';
+
+export default {
+    name: 'SpotRequestsPage',
+    components: {
+        LoaderModal,
+        MoleculeSearchBox,
+    },
+    computed: {
+        // Map Vuex state for loading status and data for spot requests
+        ...mapState('spotRequests', [
+            'isLoading',
+            'hasError',
+            'errorMessage',
+            'spotRequests',
+        ]),
+    },
+    
+    mounted() {
+        // Fetch initial spot requests data on component mount
+        this.fetchSpotRequests();
+    },
+    methods: {
+        ...mapActions('spotRequests', ['fetchSpotRequests']),
+
+        // Search by Request ID with validation for numeric input
+        searchSpotRequest(requestId) {
+            const isValidRequestId = /^[0-9]+$/.test(requestId);
+            if (!isValidRequestId) {
+                this.alertError('Please enter a valid numeric Request ID.');
+                return;
+            }
+            // Update route with search parameter for Request ID
+            this.$router.push({
+                path: this.$route.path,
+                query: { requestId: requestId },
+            });
+        },
+
+        // Generate detail URL for a specific Request ID 
+        // Not working properly
```

Comment:
its not working because in routes.js file there in no route entry with name as spot-requests.
There is spotRequest, if you want to navigate there use that.